artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java

```diff
@@ -58,13 +61,24 @@ public static void executeProcess(List<String> command, String errorMsg, Path pa
         try {
             Process process = builder.redirectErrorStream(true)
                     .directory(path.toFile())
-                    .redirectOutput(ProcessBuilder.Redirect.DISCARD)
-                    .redirectError(ProcessBuilder.Redirect.DISCARD)
                     .start();
+            StringBuilder output = new StringBuilder();
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    output.append(line).append("\n");
+                }
+            }
             process.waitFor();
+            if(process.exitValue() != 0) {
+                throw new RuntimeOperationsException(
+                        new RuntimeException("Process exited with code: " + process.exitValue()),
+                        errorMsg + " Output: " + output
+                );
+            }
             assertEquals(0, process.exitValue(), errorMsg);
         } catch (IOException | InterruptedException e) {
-            throw new RuntimeException(e);
+            throw new RuntimeOperationsException(new RuntimeException(e), errorMsg);
```

Comment:
```suggestion
            throw new RuntimeOperationsException(e, errorMsg);
```

Reply:
That's not possible in this case, the arguments required for RuntimeOperationsException are :
`RuntimeOperationsException(java.lang.RuntimeException e, String message) `

Reply:
So throw https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/RuntimeException.html#%3Cinit%3E(java.lang.String,java.lang.Throwable)